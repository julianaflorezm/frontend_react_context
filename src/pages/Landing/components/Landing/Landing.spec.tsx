import { fireEvent, render } from '@testing-library/react';
import { LandingProps, Landing } from './index';

describe('Landing Test', () => {
  let props: LandingProps;

  beforeEach(() => {
    props = {
      onUsername: jest.fn(),
      onId: jest.fn(),
    };
  });

  it('should match snapshot', () => {
    const { container } = render(<Landing {...props} />);
    expect(container).toMatchSnapshot();
  });

  it('should call a function when button is clicked', async () => {
    const { getByText } = render(<Landing {...props} />);
    const button = getByText('Ingresar');
    fireEvent.click(button);
    expect(getByText('Ingresar')).toBeTruthy();
  });

  it('should call a function when button is clicked', async () => {
    const { getByText } = render(<Landing {...props} />);
    const button = getByText('Registrarse');
    fireEvent.click(button);
    expect(getByText('Registrarse')).toBeTruthy();
  });
});
