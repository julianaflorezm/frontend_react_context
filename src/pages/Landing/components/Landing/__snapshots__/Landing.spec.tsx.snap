// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Landing Test should match snapshot 1`] = `
<div>
  <div
    class="MuiGrid-root makeStyles-position-8 MuiGrid-container MuiGrid-item MuiGrid-grid-xs-4 MuiGrid-grid-md-2"
  >
    <button
      class="MuiButtonBase-root MuiButton-root MuiButton-contained makeStyles-buttonReg-7 MuiButton-fullWidth"
      tabindex="0"
      type="button"
    >
      <span
        class="MuiButton-label"
      >
        Registrarse
      </span>
      <span
        class="MuiTouchRipple-root"
      />
    </button>
  </div>
  <div
    class="MuiGrid-root makeStyles-container-1 MuiGrid-container"
  >
    <div
      class="MuiGrid-root MuiGrid-container MuiGrid-item MuiGrid-justify-xs-center MuiGrid-grid-xs-10 MuiGrid-grid-md-6"
    >
      <div
        class="MuiGrid-root MuiGrid-item"
      >
        <div
          class="makeStyles-containerImg-2"
        >
          <img
            alt="Logo"
            class="makeStyles-img-3"
            src="bank.png"
          />
        </div>
        <p
          class="MuiTypography-root makeStyles-title-4 MuiTypography-body2 MuiTypography-colorTextSecondary MuiTypography-alignCenter"
        >
          Bienvenido a su banco
        </p>
        <p
          class="MuiTypography-root MuiTypography-body1 MuiTypography-colorTextSecondary MuiTypography-alignCenter"
        >
          Ingrese con su usuario y contrase√±a
        </p>
      </div>
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-6"
      >
        <div
          class="MuiFormControl-root MuiTextField-root makeStyles-input-5 MuiFormControl-fullWidth"
        >
          <label
            class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined"
            data-shrink="false"
          >
            Nombre de usuario
          </label>
          <div
            class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-fullWidth MuiInputBase-formControl"
          >
            <input
              aria-invalid="false"
              class="MuiInputBase-input MuiOutlinedInput-input"
              type="text"
              value=""
            />
            <fieldset
              aria-hidden="true"
              class="PrivateNotchedOutline-root-14 MuiOutlinedInput-notchedOutline"
            >
              <legend
                class="PrivateNotchedOutline-legendLabelled-16"
              >
                <span>
                  Nombre de usuario
                </span>
              </legend>
            </fieldset>
          </div>
        </div>
        <div
          class="MuiFormControl-root MuiTextField-root makeStyles-input-5 MuiFormControl-fullWidth"
        >
          <label
            class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined"
            data-shrink="false"
          >
            Clave
          </label>
          <div
            class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-fullWidth MuiInputBase-formControl"
          >
            <input
              aria-invalid="false"
              class="MuiInputBase-input MuiOutlinedInput-input"
              type="password"
              value=""
            />
            <fieldset
              aria-hidden="true"
              class="PrivateNotchedOutline-root-14 MuiOutlinedInput-notchedOutline"
            >
              <legend
                class="PrivateNotchedOutline-legendLabelled-16"
              >
                <span>
                  Clave
                </span>
              </legend>
            </fieldset>
          </div>
        </div>
        <button
          class="MuiButtonBase-root MuiButton-root MuiButton-contained makeStyles-button-6 MuiButton-fullWidth"
          tabindex="0"
          type="button"
        >
          <span
            class="MuiButton-label"
          >
            Ingresar
          </span>
          <span
            class="MuiTouchRipple-root"
          />
        </button>
      </div>
    </div>
  </div>
</div>
`;
